backend/src/services/auth.rs

```diff
@@ -0,0 +1,349 @@
+use super::prelude::*;
+
+pub type HttpClient = ContextWrapper<
```

Comment:
Can you explain what's going on with `ClientContext` here? Why do you declare the `Client` without context (`DropContextService`) and then wrap it with context? What is the reason that `Client` was not declared directly with the context?

Reply:
In theory, we can wrap any context with any context provider (for example, the one that reads\writes context from\to file, and not just warps the context and keeps it in-memory). Practically, I could also solve that with just generic types and some methods, not the whole wrapper, but since that was the part of auto-generated code, I decided to keep it this way